javav2/example_code/kendra/src/main/java/com/example/kendra/QueryIndex.java

```diff
@@ -0,0 +1,88 @@
+//snippet-sourcedescription:[QueryIndex.java demonstrates how to query an Amazon Kendra index.]
+//snippet-keyword:[SDK for Java v2]
+//snippet-keyword:[Code Sample]
+//snippet-service:[Amazon Kendra]
+//snippet-sourcetype:[full-example]
+//snippet-sourcedate:[03/10/2022]
+//snippet-sourceauthor:[scmacdon - aws]
+
+/*
+   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0
+*/
+
+
+package com.example.kendra;
+
+// snippet-start:[kendra.java2.query.index.import]
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.kendra.KendraClient;
+import software.amazon.awssdk.services.kendra.model.QueryRequest;
+import software.amazon.awssdk.services.kendra.model.QueryResultType;
+import software.amazon.awssdk.services.kendra.model.QueryResponse;
+import software.amazon.awssdk.services.kendra.model.QueryResultItem;
+import software.amazon.awssdk.services.kendra.model.KendraException;
+import java.util.List;
+// snippet-end:[kendra.java2.query.index.import]
+
+/**
+ * To run this Java V2 code example, ensure that you have set up your development environment, including your credentials.

```

Comment:
Before you run this Java V2 code example, set up your development environment, including your credentials.